modules/unsupported/dggs/dggs-core/src/main/java/org/geotools/dggs/datastore/DGGSFeatureSourceImpl.java

```diff
@@ -223,6 +224,9 @@ public SimpleFeatureCollection getFeatures(Query query) throws IOException {
         } else if (delegate.getDataStore() instanceof JDBCDataStore) {
             result = new AggregatorCollection(
                     result, (JDBCDataStore) delegate.getDataStore(), new Query(split.pre), getSchema());
+        } else if (delegate.getDataStore() instanceof Wrapper w) {
+            JDBCDataStore jdbcDataStore = w.unwrap(JDBCDataStore.class);
```

Comment:
Being a wrapper does not mean that it will successfuly unwrap to a JDBCDataStore. I would add checks.